### PR TITLE
infer task type

### PR DESCRIPTION
**Pull request by benisraeldan**
infer task type from dataset

### CODE REVIEW

File: deepchecks/base/dataset.py [hunk 1]
```diff
@@ -420,6 +441,20 @@ def train_test_split(self,
...
 
+    @staticmethod
+    def _infer_label_type(
+            label_col: pd.Series,
+            max_categorical_ratio: float,
+            max_categories: int,
+            max_float_categories: int
+    ):
+        if not is_numeric_dtype(label_col):
+            return 'classification label'
```
Comment (ItayGabbay) [inline deepchecks/base/dataset.py:452]: I would remove the "label" word from the type.

File: deepchecks/base/dataset.py [hunk 2]
```diff
@@ -420,6 +441,20 @@ def train_test_split(self,
...
+        elif is_categorical(label_col, max_categorical_ratio, max_categories, max_float_categories):
+            return 'classification label'
+        else:
+            return 'regression label'
```
Comment (ItayGabbay) [inline deepchecks/base/dataset.py:456]: I just hope it's a good heuristic....

File: deepchecks/base/dataset.py (hunk 1 above)
Comment (benisraeldan) [reply deepchecks/base/dataset.py:452]: @nirhutnik - having it just classification/regression is weird for a label type

Comment (ItayGabbay) [reply deepchecks/base/dataset.py:452]: at least let's replace the space with _

File: deepchecks/base/dataset.py (hunk 2 above)
Comment (noamzbr) [reply deepchecks/base/dataset.py]: I'm also not sure about it, but it's better than using something like https://scikit-learn.org/stable/modules/generated/sklearn.utils.multiclass.type_of_target.html#sklearn.utils.multiclass.type_of_target - this will for e.g. infer a "test score" that is ints from 0 to 100 as "classification" although it's clearly regression. 

We're open to suggestions

**Review by ItayGabbay — APPROVED** [review]
